service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt

```diff
@@ -67,9 +67,20 @@ fun Database.Transaction.deleteAbsencesCreatedFromQuestionnaire(questionnaireId:
         .execute()
 }
 
-fun Database.Transaction.clearOldAbsencesExcludingFreeAbsences(childDatePairs: List<Pair<ChildId, LocalDate>>) {
+/**
+ * A citizen is allowed to edit:
+ * - absences added by a citizen
+ * - other than FREE_ABSENCE
+ */
+fun Database.Transaction.clearOldCitizenEditableAbsences(childDatePairs: List<Pair<ChildId, LocalDate>>) {
     val batch = prepareBatch(
-        "DELETE FROM absence WHERE child_id = :childId AND date = :date AND absence_type <> 'FREE_ABSENCE'::absence_type"
+        """
+DELETE FROM absence 
+WHERE child_id = :childId
+AND date = :date
+AND absence_type <> 'FREE_ABSENCE'::absence_type
+AND (SELECT evaka_user.type = 'CITIZEN' FROM evaka_user WHERE evaka_user.id = absence.modified_by)
```

Comment:
The same query is used when employee creates reservations for children in the week view. So, it effectively prevents creating a reservation when there is an absence on the same day. The save operation just silently ignores the saved attendance unless the absence is deleted first. This seems fine to me, but the UX is slightly confusing at first.
<img width="176" alt="kuva" src="https://user-images.githubusercontent.com/1176169/160583378-f0f140dd-295a-45db-bec9-395e36843949.png">

Reply:
Let's check this UX separately